dts/arm/ambiq/ambiq_apollo4p.dtsi

```diff
@@ -98,6 +98,87 @@
 			compatible = "ambiq,apollo4-pinctrl";
 			reg = <0x40010000 0x800>;
 		};
+		spi0: spi@40050000 {
```

Comment:
this is sharing the same address as i2c nodes in https://github.com/zephyrproject-rtos/zephyr/pull/61330/files#diff-6cb7ed21e635c7d7f989d07541c7952755a882454562121df424c417aa1c41fe, so what you should do it to have a single node, and change the compatible as needed to define the actual configuration. There are some examples in tree.

Reply:
As I said:

> Also Ambiq IOM I2C and SPI are using the same peripherals. The change will be needed after SPI or I2C (https://github.com/zephyrproject-rtos/zephyr/pull/61330) driver will be merged.

After `I2C` merge I will change the `SPI` nodes, so we will have 4 `I2C` nodes and 4 `SPI` nodes that can be changed by the user.

Reply:
This is now changed.